{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Astrobotany project: Finding patterns from space data using statistical method"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So far, from first milestone of the project, objective of the project(name of the project) was defined with help from Choi et.al paper. Then, data exploration was done with BRIC19 data by using R. Few of concerns and possible suggestions with data was discussed. Possible statistical method such as unsupervised learning(e.g. clustering) and deep learning(e.g. NLP based neural networks) was suggested.\n",
    "After formal discussion with first milestone that I have done, it is decided that new data from NASA Gene lab GLDS-120 will be used to perform the analysis. Therefore, process that was done with first mileston, such as, data preprocessing and data exploration will be performed again before building the main model. This time Python will be used for the analysis."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Study Description of GLDS-120: \n",
    "\n",
    "Experimentation on the International Space Station has reached the stage where repeated and nuanced transcriptome studies are beginning to illuminate the **structural and metabolic differences between plants grown in space compared to plants on the Earth.** \n",
    "\n",
    "Genes that are important in setting up the spaceflight responses are being identified; their role in spaceflight physiological adaptation are increasingly understood, and the fact that different genotypes adapt differently is recognized. However, the basic question of whether these spaceflight responses are required for survival has yet to be posed, and the fundamental notion that spaceflight responses may be non-adaptive has yet to be explored.\n",
    "Therefore the experiments presented here were designed to ask **if portions of the plant spaceflight response can be genetically removed without causing loss of spaceflight survival and without causing increased stress responses.**\n",
    "\n",
    "The CARA experiment compared the spaceflight transcriptome responses of two Arabidopsis ecotypes, Col-0 and WS, as well as that of a PhyD mutant of Col-0. When grown with the ambient light of the ISS, phyD displayed a significantly reduced spaceflight transcriptome response compared to Col-0, **suggesting that altering the activity of a single gene can actually improve spaceflight adaptation by reducing the transcriptome cost of physiological adaptation.** The WS genotype showed an even simpler spaceflight transcriptome response in the ambient light of the ISS, more **broadly indicating that the plant genotype can be manipulated to reduce the transcriptome cost of plant physiological adaptation to spaceflight and suggesting that genetic manipulation might further reduce, or perhaps eliminate the metabolic cost of spaceflight adaptation.** \n",
    "\n",
    "When plants were germinated and then left in the dark on the ISS, the WS genotype actually mounted a larger transcriptome response than Col-0, suggesting that the in-space light environment affects physiological adaptation, which further **implies that manipulating the local habitat can also substantially impact the metabolic cost of spaceflight adaptation**."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "'Normalized_counts' data from GLDS-120 will be mainly used and  'Array_Genediff_pilot','RNAseq_Genediff_pilot' and 'RNAseq_Isoformsdiff_pilot' will be used as supplemantary data. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Objective of this project will be remained same:\n",
    "Finding patterns from space data using statistical method"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alex-Seo \n",
      "\n",
      "CPython 3.6.2\n",
      "IPython 6.1.0\n",
      "\n",
      "matplotlib 3.2.0\n",
      "numpy 1.18.1\n",
      "pandas 1.0.1\n"
     ]
    }
   ],
   "source": [
    "%load_ext watermark\n",
    "%watermark -a 'Alex-Seo' -v -p matplotlib,numpy,pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Data Exploration\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random as rnd\n",
    "\n",
    "# visualization\n",
    "import altair as alt\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "nc_all = pd.read_csv(\"GLDS-120_rna_seq_Normalized_Counts.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RNA_seq</th>\n",
       "      <th>Atha_Col-0-PhyD_root_FLT_Alight_Rep1_GSM2493783</th>\n",
       "      <th>Atha_Col-0-PhyD_root_FLT_Alight_Rep2_GSM2493784</th>\n",
       "      <th>Atha_Col-0-PhyD_root_FLT_Alight_Rep3_GSM2493785</th>\n",
       "      <th>Atha_Col-0-PhyD_root_FLT_dark_Rep1_GSM2493792</th>\n",
       "      <th>Atha_Col-0-PhyD_root_FLT_dark_Rep2_GSM2493793</th>\n",
       "      <th>Atha_Col-0-PhyD_root_FLT_dark_Rep3_GSM2493794</th>\n",
       "      <th>Atha_Col-0-PhyD_root_GC_Alight_Rep1_GSM2493765</th>\n",
       "      <th>Atha_Col-0-PhyD_root_GC_Alight_Rep2_GSM2493766</th>\n",
       "      <th>Atha_Col-0-PhyD_root_GC_Alight_Rep3_GSM2493767</th>\n",
       "      <th>...</th>\n",
       "      <th>Atha_Ws_root_FLT_Alight_Rep3_GSM2493782</th>\n",
       "      <th>Atha_Ws_root_FLT_dark_Rep1_GSM2493789</th>\n",
       "      <th>Atha_Ws_root_FLT_dark_Rep2_GSM2493790</th>\n",
       "      <th>Atha_Ws_root_FLT_dark_Rep3_GSM2493791</th>\n",
       "      <th>Atha_Ws_root_GC_Alight_Rep1_GSM2493762</th>\n",
       "      <th>Atha_Ws_root_GC_Alight_Rep2_GSM2493763</th>\n",
       "      <th>Atha_Ws_root_GC_Alight_Rep3_GSM2493764</th>\n",
       "      <th>Atha_Ws_root_GC_dark_Rep1_GSM2493771</th>\n",
       "      <th>Atha_Ws_root_GC_dark_Rep2_GSM2493772</th>\n",
       "      <th>Atha_Ws_root_GC_dark_Rep3_GSM2493773</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>116.888298</td>\n",
       "      <td>132.379479</td>\n",
       "      <td>212.743795</td>\n",
       "      <td>182.011159</td>\n",
       "      <td>37.451300</td>\n",
       "      <td>36.191007</td>\n",
       "      <td>97.652423</td>\n",
       "      <td>193.262178</td>\n",
       "      <td>105.192824</td>\n",
       "      <td>...</td>\n",
       "      <td>88.250619</td>\n",
       "      <td>53.594919</td>\n",
       "      <td>139.944149</td>\n",
       "      <td>149.854946</td>\n",
       "      <td>62.133158</td>\n",
       "      <td>109.261341</td>\n",
       "      <td>80.132301</td>\n",
       "      <td>21.675591</td>\n",
       "      <td>69.483039</td>\n",
       "      <td>115.772776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AT1G01020</td>\n",
       "      <td>309.753663</td>\n",
       "      <td>275.664706</td>\n",
       "      <td>271.519734</td>\n",
       "      <td>270.261115</td>\n",
       "      <td>369.703512</td>\n",
       "      <td>323.787491</td>\n",
       "      <td>321.676426</td>\n",
       "      <td>235.522120</td>\n",
       "      <td>331.126841</td>\n",
       "      <td>...</td>\n",
       "      <td>242.874452</td>\n",
       "      <td>212.480431</td>\n",
       "      <td>247.649370</td>\n",
       "      <td>193.853179</td>\n",
       "      <td>277.896901</td>\n",
       "      <td>235.501416</td>\n",
       "      <td>250.294311</td>\n",
       "      <td>217.427950</td>\n",
       "      <td>233.272214</td>\n",
       "      <td>244.534539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AT1G01030</td>\n",
       "      <td>15.605457</td>\n",
       "      <td>31.582754</td>\n",
       "      <td>33.380759</td>\n",
       "      <td>39.331565</td>\n",
       "      <td>66.818571</td>\n",
       "      <td>114.980428</td>\n",
       "      <td>40.688977</td>\n",
       "      <td>79.069498</td>\n",
       "      <td>69.055665</td>\n",
       "      <td>...</td>\n",
       "      <td>43.629223</td>\n",
       "      <td>17.030337</td>\n",
       "      <td>48.459228</td>\n",
       "      <td>63.638014</td>\n",
       "      <td>33.698318</td>\n",
       "      <td>58.330341</td>\n",
       "      <td>33.802204</td>\n",
       "      <td>48.068188</td>\n",
       "      <td>25.012820</td>\n",
       "      <td>34.297481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AT1G01040</td>\n",
       "      <td>87.827413</td>\n",
       "      <td>66.439742</td>\n",
       "      <td>59.069120</td>\n",
       "      <td>65.637198</td>\n",
       "      <td>118.481842</td>\n",
       "      <td>83.514259</td>\n",
       "      <td>91.966383</td>\n",
       "      <td>101.624349</td>\n",
       "      <td>58.173340</td>\n",
       "      <td>...</td>\n",
       "      <td>119.232764</td>\n",
       "      <td>123.953587</td>\n",
       "      <td>118.829533</td>\n",
       "      <td>163.138572</td>\n",
       "      <td>119.058956</td>\n",
       "      <td>76.548721</td>\n",
       "      <td>116.494389</td>\n",
       "      <td>97.803263</td>\n",
       "      <td>132.859933</td>\n",
       "      <td>151.702383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AT1G01050</td>\n",
       "      <td>1381.723141</td>\n",
       "      <td>1260.158962</td>\n",
       "      <td>1304.779193</td>\n",
       "      <td>1286.505645</td>\n",
       "      <td>940.910219</td>\n",
       "      <td>1425.881579</td>\n",
       "      <td>1121.849300</td>\n",
       "      <td>1727.003728</td>\n",
       "      <td>1415.289380</td>\n",
       "      <td>...</td>\n",
       "      <td>1545.700537</td>\n",
       "      <td>1241.428582</td>\n",
       "      <td>1111.929045</td>\n",
       "      <td>868.365582</td>\n",
       "      <td>1132.979712</td>\n",
       "      <td>1181.454980</td>\n",
       "      <td>1323.852859</td>\n",
       "      <td>1492.290175</td>\n",
       "      <td>1205.715025</td>\n",
       "      <td>966.320255</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     RNA_seq  Atha_Col-0-PhyD_root_FLT_Alight_Rep1_GSM2493783  \\\n",
       "0  AT1G01010                                       116.888298   \n",
       "1  AT1G01020                                       309.753663   \n",
       "2  AT1G01030                                        15.605457   \n",
       "3  AT1G01040                                        87.827413   \n",
       "4  AT1G01050                                      1381.723141   \n",
       "\n",
       "   Atha_Col-0-PhyD_root_FLT_Alight_Rep2_GSM2493784  \\\n",
       "0                                       132.379479   \n",
       "1                                       275.664706   \n",
       "2                                        31.582754   \n",
       "3                                        66.439742   \n",
       "4                                      1260.158962   \n",
       "\n",
       "   Atha_Col-0-PhyD_root_FLT_Alight_Rep3_GSM2493785  \\\n",
       "0                                       212.743795   \n",
       "1                                       271.519734   \n",
       "2                                        33.380759   \n",
       "3                                        59.069120   \n",
       "4                                      1304.779193   \n",
       "\n",
       "   Atha_Col-0-PhyD_root_FLT_dark_Rep1_GSM2493792  \\\n",
       "0                                     182.011159   \n",
       "1                                     270.261115   \n",
       "2                                      39.331565   \n",
       "3                                      65.637198   \n",
       "4                                    1286.505645   \n",
       "\n",
       "   Atha_Col-0-PhyD_root_FLT_dark_Rep2_GSM2493793  \\\n",
       "0                                      37.451300   \n",
       "1                                     369.703512   \n",
       "2                                      66.818571   \n",
       "3                                     118.481842   \n",
       "4                                     940.910219   \n",
       "\n",
       "   Atha_Col-0-PhyD_root_FLT_dark_Rep3_GSM2493794  \\\n",
       "0                                      36.191007   \n",
       "1                                     323.787491   \n",
       "2                                     114.980428   \n",
       "3                                      83.514259   \n",
       "4                                    1425.881579   \n",
       "\n",
       "   Atha_Col-0-PhyD_root_GC_Alight_Rep1_GSM2493765  \\\n",
       "0                                       97.652423   \n",
       "1                                      321.676426   \n",
       "2                                       40.688977   \n",
       "3                                       91.966383   \n",
       "4                                     1121.849300   \n",
       "\n",
       "   Atha_Col-0-PhyD_root_GC_Alight_Rep2_GSM2493766  \\\n",
       "0                                      193.262178   \n",
       "1                                      235.522120   \n",
       "2                                       79.069498   \n",
       "3                                      101.624349   \n",
       "4                                     1727.003728   \n",
       "\n",
       "   Atha_Col-0-PhyD_root_GC_Alight_Rep3_GSM2493767  ...  \\\n",
       "0                                      105.192824  ...   \n",
       "1                                      331.126841  ...   \n",
       "2                                       69.055665  ...   \n",
       "3                                       58.173340  ...   \n",
       "4                                     1415.289380  ...   \n",
       "\n",
       "   Atha_Ws_root_FLT_Alight_Rep3_GSM2493782  \\\n",
       "0                                88.250619   \n",
       "1                               242.874452   \n",
       "2                                43.629223   \n",
       "3                               119.232764   \n",
       "4                              1545.700537   \n",
       "\n",
       "   Atha_Ws_root_FLT_dark_Rep1_GSM2493789  \\\n",
       "0                              53.594919   \n",
       "1                             212.480431   \n",
       "2                              17.030337   \n",
       "3                             123.953587   \n",
       "4                            1241.428582   \n",
       "\n",
       "   Atha_Ws_root_FLT_dark_Rep2_GSM2493790  \\\n",
       "0                             139.944149   \n",
       "1                             247.649370   \n",
       "2                              48.459228   \n",
       "3                             118.829533   \n",
       "4                            1111.929045   \n",
       "\n",
       "   Atha_Ws_root_FLT_dark_Rep3_GSM2493791  \\\n",
       "0                             149.854946   \n",
       "1                             193.853179   \n",
       "2                              63.638014   \n",
       "3                             163.138572   \n",
       "4                             868.365582   \n",
       "\n",
       "   Atha_Ws_root_GC_Alight_Rep1_GSM2493762  \\\n",
       "0                               62.133158   \n",
       "1                              277.896901   \n",
       "2                               33.698318   \n",
       "3                              119.058956   \n",
       "4                             1132.979712   \n",
       "\n",
       "   Atha_Ws_root_GC_Alight_Rep2_GSM2493763  \\\n",
       "0                              109.261341   \n",
       "1                              235.501416   \n",
       "2                               58.330341   \n",
       "3                               76.548721   \n",
       "4                             1181.454980   \n",
       "\n",
       "   Atha_Ws_root_GC_Alight_Rep3_GSM2493764  \\\n",
       "0                               80.132301   \n",
       "1                              250.294311   \n",
       "2                               33.802204   \n",
       "3                              116.494389   \n",
       "4                             1323.852859   \n",
       "\n",
       "   Atha_Ws_root_GC_dark_Rep1_GSM2493771  Atha_Ws_root_GC_dark_Rep2_GSM2493772  \\\n",
       "0                             21.675591                             69.483039   \n",
       "1                            217.427950                            233.272214   \n",
       "2                             48.068188                             25.012820   \n",
       "3                             97.803263                            132.859933   \n",
       "4                           1492.290175                           1205.715025   \n",
       "\n",
       "   Atha_Ws_root_GC_dark_Rep3_GSM2493773  \n",
       "0                            115.772776  \n",
       "1                            244.534539  \n",
       "2                             34.297481  \n",
       "3                            151.702383  \n",
       "4                            966.320255  \n",
       "\n",
       "[5 rows x 37 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Summary\n",
    "nc_all.head()\n",
    "#print(nc_all.columns.values)\n",
    "#nc_all.describe(include='all')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RNA_seq</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Light_Setting</th>\n",
       "      <th>Rep1</th>\n",
       "      <th>Rep2</th>\n",
       "      <th>Rep3</th>\n",
       "      <th>Location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>phyD</td>\n",
       "      <td>Light</td>\n",
       "      <td>116.888298</td>\n",
       "      <td>132.379479</td>\n",
       "      <td>212.743795</td>\n",
       "      <td>FLT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>phyD</td>\n",
       "      <td>Dark</td>\n",
       "      <td>182.011159</td>\n",
       "      <td>37.451300</td>\n",
       "      <td>36.191007</td>\n",
       "      <td>FLT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>phyD</td>\n",
       "      <td>Light</td>\n",
       "      <td>97.652423</td>\n",
       "      <td>193.262178</td>\n",
       "      <td>105.192824</td>\n",
       "      <td>GC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>phyD</td>\n",
       "      <td>Dark</td>\n",
       "      <td>154.933696</td>\n",
       "      <td>88.312487</td>\n",
       "      <td>136.879249</td>\n",
       "      <td>GC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>col0</td>\n",
       "      <td>Light</td>\n",
       "      <td>119.461406</td>\n",
       "      <td>116.426029</td>\n",
       "      <td>140.074158</td>\n",
       "      <td>FLT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     RNA_seq Genotype Light_Setting        Rep1        Rep2        Rep3  \\\n",
       "0  AT1G01010     phyD         Light  116.888298  132.379479  212.743795   \n",
       "1  AT1G01010     phyD          Dark  182.011159   37.451300   36.191007   \n",
       "2  AT1G01010     phyD         Light   97.652423  193.262178  105.192824   \n",
       "3  AT1G01010     phyD          Dark  154.933696   88.312487  136.879249   \n",
       "4  AT1G01010     col0         Light  119.461406  116.426029  140.074158   \n",
       "\n",
       "  Location  \n",
       "0      FLT  \n",
       "1      FLT  \n",
       "2       GC  \n",
       "3       GC  \n",
       "4      FLT  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "###Data reconstruction, Make more variables and set the output(Y) variable\n",
    "###You can not use melt() from pandas since there are 3 rep for each segment\n",
    "\n",
    "###replicate each row 12 times for same RNA_seq\n",
    "newnc = pd.DataFrame(np.repeat(nc_all.values,12,axis=0))\n",
    "#newnc.colums = nc_all.columns\n",
    "\n",
    "###Create Genotype variable\n",
    "GT = pd.DataFrame(\n",
    "{\"Genotype\" : ['phyD', 'phyD', 'phyD', 'phyD',\n",
    "               'col0', 'col0','col0','col0',\n",
    "               'WS', 'WS', 'WS', 'WS']})\n",
    "GT = pd.concat([GT]*len(nc_all)).reset_index().drop(columns=['index'])\n",
    "\n",
    "###Create Light setting variable\n",
    "\n",
    "LS = pd.DataFrame(\n",
    "{\"Light_Setting\" : ['Light' ,'Dark']})\n",
    "LS = pd.concat([LS]*6*len(nc_all)).reset_index().drop(columns=['index'])\n",
    "\n",
    "###create Location variable\n",
    "LC = pd.DataFrame(\n",
    "{\"Location\" : ['FLT' ,'FLT', 'GC', 'GC']})\n",
    "LC = pd.concat([LC]*3*len(nc_all)).reset_index().drop(columns=['index'])\n",
    "\n",
    "###Transpose fpkm values (3 values)\n",
    "ncworna = nc_all.drop(columns = ['RNA_seq'])\n",
    "D = pd.DataFrame(columns=['Rep1', 'Rep2', 'Rep3'])\n",
    "\n",
    "for i in range(len(ncworna)):\n",
    "    s = pd.Series(ncworna.iloc[i]) #Make each row a pandas series\n",
    "    a = np.reshape(s.to_numpy(),(12,3)) #Change Series to numpy and reshape it\n",
    "    #Make each reshaped arrays into dataframe, this one dataframe represent one RNA sequence\n",
    "    d = pd.DataFrame(a, columns=['Rep1', 'Rep2', 'Rep3'])\n",
    "    D = pd.concat([D,d]) #concat each dataframe\n",
    "\n",
    "D = D.reset_index().drop(columns=['index'])#Reset index\n",
    "###Add and drop variables\n",
    "newnc = newnc.drop(columns=range(1,37))\n",
    "\n",
    "newnc[\"Genotype\"] = GT\n",
    "newnc[\"Light_Setting\"] = LS\n",
    "\n",
    "newnc = pd.concat([newnc, D], axis = 1)\n",
    "\n",
    "newnc[\"Location\"] = LC\n",
    "\n",
    "#Rename column\n",
    "newnc = newnc.rename(columns = {0 : \"RNA_seq\"})\n",
    "#Overview of newly constructed dataset\n",
    "newnc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RNA_seq</th>\n",
       "      <th>Rep1</th>\n",
       "      <th>Rep2</th>\n",
       "      <th>Rep3</th>\n",
       "      <th>Location</th>\n",
       "      <th>Genotype_WS</th>\n",
       "      <th>Genotype_col0</th>\n",
       "      <th>Genotype_phyD</th>\n",
       "      <th>Light_Setting_Dark</th>\n",
       "      <th>Light_Setting_Light</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>116.888298</td>\n",
       "      <td>132.379479</td>\n",
       "      <td>212.743795</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>182.011159</td>\n",
       "      <td>37.451300</td>\n",
       "      <td>36.191007</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>97.652423</td>\n",
       "      <td>193.262178</td>\n",
       "      <td>105.192824</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>154.933696</td>\n",
       "      <td>88.312487</td>\n",
       "      <td>136.879249</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>119.461406</td>\n",
       "      <td>116.426029</td>\n",
       "      <td>140.074158</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     RNA_seq        Rep1        Rep2        Rep3  Location  Genotype_WS  \\\n",
       "0  AT1G01010  116.888298  132.379479  212.743795         1            0   \n",
       "1  AT1G01010  182.011159   37.451300   36.191007         1            0   \n",
       "2  AT1G01010   97.652423  193.262178  105.192824         0            0   \n",
       "3  AT1G01010  154.933696   88.312487  136.879249         0            0   \n",
       "4  AT1G01010  119.461406  116.426029  140.074158         1            0   \n",
       "\n",
       "   Genotype_col0  Genotype_phyD  Light_Setting_Dark  Light_Setting_Light  \n",
       "0              0              1                   0                    1  \n",
       "1              0              1                   1                    0  \n",
       "2              0              1                   0                    1  \n",
       "3              0              1                   1                    0  \n",
       "4              1              0                   0                    1  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Applying models using newly constructed labeled dataset\n",
    "\n",
    "#Label Encoding - One hot encoding on categorical predictors for modeling purpose\n",
    "ncwd = pd.get_dummies(newnc, columns = [\"Genotype\"]) #drop_first =True\n",
    "ncwd = pd.get_dummies(ncwd, columns = ['Light_Setting'])\n",
    "\n",
    "clean_target = {\"Location\": {\"FLT\": 1, \"GC\": 0}}\n",
    "ncwd.replace(clean_target, inplace = True)\n",
    "\n",
    "ncwd.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\LG\\Anaconda3\\lib\\site-packages\\sklearn\\ensemble\\weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.\n",
      "  from numpy.core.umath_tests import inner1d\n"
     ]
    }
   ],
   "source": [
    "#Pacakages\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.svm import SVC, LinearSVC\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "from sklearn.naive_bayes import GaussianNB\n",
    "from sklearn.linear_model import Perceptron\n",
    "from sklearn.linear_model import SGDClassifier\n",
    "from sklearn.tree import DecisionTreeClassifier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "230265 57567\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(230265, 8)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Model 1 - Start with a basics: Since the target variables are binary, Apply Logistic Regression for classification\n",
    "\n",
    "#Shuffle, then divide it into training and test dataset\n",
    "y = ncwd['Location'].values\n",
    "X = ncwd.iloc[:, [1,2,3,5,6,7,8,9]].values\n",
    "#Shuffle\n",
    "indices = np.arange(X.shape[0])\n",
    "rng = np.random.RandomState(123)\n",
    "permuted_indices = rng.permutation(indices)\n",
    "#Split - 80(Train) : 20(Test)\n",
    "train_size = int(0.8*X.shape[0])\n",
    "test_size = X.shape[0] - (train_size)\n",
    "print(train_size, test_size)\n",
    "\n",
    "train_ind = permuted_indices[:train_size]\n",
    "test_ind = permuted_indices[train_size:]\n",
    "\n",
    "X_train, y_train = X[train_ind], y[train_ind]\n",
    "X_test, y_test = X[test_ind], y[test_ind]\n",
    "\n",
    "X_train.shape\n",
    "\n",
    "## No more preprocessing needed such as, scaling and normalization. - else use pipelines form sckitlearn\n",
    "## All features, Genotype, Lightsetting and fpkm shouldn't be modified in anyways since first two are categorical variables\n",
    "## and fpkm values have meanings itself"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "50.19\n"
     ]
    }
   ],
   "source": [
    "#Logistic Regression\n",
    "\n",
    "logreg = LogisticRegression()\n",
    "logreg.fit(X_train, y_train)\n",
    "#y_pred = logreg.predict(X_test)\n",
    "acc_log = round(logreg.score(X_test, y_test) * 100, 2)\n",
    "print(acc_log)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Additional Machine Learning Models\n",
    "#SVM\n",
    "#Takes toooo much time to compile\n",
    "svc = SVC()\n",
    "svc.fit(X_train, y_train)\n",
    "#y_pred = svc.predict(X_test)\n",
    "acc_svc = round(svc.score(X_test, y_test) * 100, 2)\n",
    "print(acc_svc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "56.35\n"
     ]
    }
   ],
   "source": [
    "#KNN\n",
    "\n",
    "knn = KNeighborsClassifier(n_neighbors = 3)\n",
    "knn.fit(X_train, y_train)\n",
    "#y_pred = knn.predict(X_test)\n",
    "acc_knn = round(knn.score(X_test, y_test) * 100, 2)\n",
    "print(acc_knn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "70.07\n"
     ]
    }
   ],
   "source": [
    "# Random Forest\n",
    "\n",
    "random_forest = RandomForestClassifier(n_estimators=100)\n",
    "random_forest.fit(X_train, y_train)\n",
    "#y_pred = random_forest.predict(X_test)\n",
    "acc_random_forest = round(random_forest.score(X_test, y_test) * 100, 2)\n",
    "print(acc_random_forest)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
