---
title: "NASA exp1"
author: "Alex Seo"
date: "December 8, 2019"
output: pdf_document
header-includes:
- \usepackage{fancyhdr}
- \pagestyle{fancy}
- \fancyhf{}
- \rhead{Seo}
- \lhead{NASA}
- \cfoot{\thepage}
---

<!-- R chunk setup -->
```{r setup, include=FALSE}
# global chunk options
knitr::opts_chunk$set(echo = TRUE, comment = NA)

# load and attach add-on packages here
library(knitr)
library(qcc)
library(tidyverse)
library(leaps)
library(MVA)
library(MASS)
library(ggfortify)
```

<!-- Page Break -->
\newpage

#Astrobotany project: Finding patterns from space data using statistical method

> So far, from first milestone of the project, objective of the project(name of the project) was defined with help from Choi et.al paper. Then, data exploration was done with BRIC19 data by using R. Few of concerns and possible suggestions with data was discussed. Possible statistical method such as unsupervised learning(e.g. clustering) and deep learning(e.g. NLP based neural networks) was suggested.

> After formal discussion with first milestone that I have done, it is decided that new data from NASA Gene lab GLDS-120 will be used to perform the analysis. Therefore, process that was done with first mileston, such as, data preprocessing and data exploration will be performed again before building the main model. 

##Study Description of GLDS-120: 

Experimentation on the International Space Station has reached the stage where repeated and nuanced transcriptome studies are beginning to illuminate the **structural and metabolic differences between plants grown in space compared to plants on the Earth.** 

Genes that are important in setting up the spaceflight responses are being identified; their role in spaceflight physiological adaptation are increasingly understood, and the fact that different genotypes adapt differently is recognized. However, the basic question of whether these spaceflight responses are required for survival has yet to be posed, and the fundamental notion that spaceflight responses may be non-adaptive has yet to be explored.
Therefore the experiments presented here were designed to ask **if portions of the plant spaceflight response can be genetically removed without causing loss of spaceflight survival and without causing increased stress responses.**

The CARA experiment compared the spaceflight transcriptome responses of two Arabidopsis ecotypes, Col-0 and WS, as well as that of a PhyD mutant of Col-0. When grown with the ambient light of the ISS, phyD displayed a significantly reduced spaceflight transcriptome response compared to Col-0, **suggesting that altering the activity of a single gene can actually improve spaceflight adaptation by reducing the transcriptome cost of physiological adaptation.** The WS genotype showed an even simpler spaceflight transcriptome response in the ambient light of the ISS, more **broadly indicating that the plant genotype can be manipulated to reduce the transcriptome cost of plant physiological adaptation to spaceflight and suggesting that genetic manipulation might further reduce, or perhaps eliminate the metabolic cost of spaceflight adaptation.** 

When plants were germinated and then left in the dark on the ISS, the WS genotype actually mounted a larger transcriptome response than Col-0, suggesting that the in-space light environment affects physiological adaptation, which further **implies that manipulating the local habitat can also substantially impact the metabolic cost of spaceflight adaptation**.

- 'Normalized_counts' data from GLDS-120 will be mainly used and  'Array_Genediff_pilot','RNAseq_Genediff_pilot' and 'RNAseq_Isoformsdiff_pilot' will be used as supplemantary data. Normalized counts were expressed in terms of FPKM values (fragments per kilobase of transcript per million mapped fragments). FPKM is directly proportional to abundance of the transcript.

- Objective of this project will be remained same: Finding patterns from space data using statistical method

```{r}
nasa<-read.csv("GLDS-120_rna_seq_Normalized_Counts.csv")
##head(nasa)
##summary(nasa)
#Dividing dataset 
#col0-phyD and col0 and WS
rna_seq<-nasa[,1]
col0_phyD_root_flight<-nasa[,c(1:7)]
col0_phyD_root_GC<-nasa[,c(1,8:13)]
col0_root_flight<-nasa[,c(1,14:19)]
col0_root_GC<-nasa[,c(1,20:25)]
WS_root_flight<-nasa[,c(1,26:31)]
WS_root_GC<-nasa[,c(1,32:37)]
#Species
col0_phyD<-cbind(col0_phyD_root_flight,col0_phyD_root_GC[,-1])
col0<-cbind(col0_root_flight,col0_root_GC[,-1])
WS<-cbind(WS_root_flight,WS_root_GC[,-1])
```

```{r}
##TO make it easir to read
colnames(col0_phyD)<-c('A(1)','B(2)','C(3)','D(4)','E(5)','F(6)','G(7)','H(8)','I(9)','J(10)','K(11)','L(12)','M(13)')

##Outliers from clustering
#col0_phyD[which(km3$cluster==2),]
##row Num : 12958,23875
col0_phyD<-col0_phyD[-c(12958,23875),]

##remaining outliers from pca
#which(z1>100000)
##row Num : 861
col0_phyD<-col0_phyD[-861,]
```

> Outliers analysis

```{r}
#PCA
col0_phyD.pca<-princomp(col0_phyD[,-1])
summary(col0_phyD.pca)
z1<-col0_phyD.pca$scores[,1]
z2<-col0_phyD.pca$scores[,2]
z<-cbind(z1,z2)

pcaplot <- data.frame(PC1 = z1, PC2 = z2)

ggplot(pcaplot, aes(x = PC1, y = PC2)) +
  geom_point()

eigvals<-col0_phyD.pca$sdev^2
k<-length(eigvals)
plot(1:k,eigvals,type="b",xlab="i",ylab=expression(lambda[i]))

##PCAbiplot
ggplot2::autoplot(col0_phyD.pca,label = F, loadings.label = T)
```

> From the PCA plot

```{r}
##kmeans
n<-dim(z)[1]; k<-6
wss<- rep(0,k); xm<-apply(z,2,mean)
for(i in 1:n){
  wss[1]<- wss[1]+sum((z[i,]-xm)^2)
}
for(i in 2:k){
  model<-kmeans(z,i)
  wss[i]<-sum(model$withinss)
}
plot(1:k,wss,type="b",xlab="Number of clusters", ylab="Within cluster sum of squares",main="Screeplot")

k<-4
km4<-kmeans(z,k)

df <- data.frame(PC1 = z1, PC2 = z2, cluster = factor(km4$cluster))

ggplot(df, aes(x = PC1, y = PC2, color = cluster)) +
  geom_point()


length(col0_phyD[which(km4$cluster==1),1])

length(col0_phyD[which(km4$cluster==2),1])

length(col0_phyD[which(km4$cluster==3),1])

length(col0_phyD[which(km4$cluster==4),1])

#length(phyD_c1[which(km3$cluster==1),1])

#length(phyD_c1[which(km3$cluster==2),1])

#length(phyD_c1[which(km3$cluster==3),1])
```

```{r}
phyD_c1<-col0_phyD[which(km4$cluster==1),]
#PCA
col0_phyD1.pca<-princomp(phyD_c1[,-1])
summary(col0_phyD1.pca)
z1<-col0_phyD1.pca$scores[,1]
z2<-col0_phyD1.pca$scores[,2]
z<-cbind(z1,z2)

pcaplot <- data.frame(PC1 = z1, PC2 = z2)

ggplot(pcaplot, aes(x = PC1, y = PC2)) +
  geom_point()

eigvals<-col0_phyD1.pca$sdev^2
k<-length(eigvals)
plot(1:k,eigvals,type="b",xlab="i",ylab=expression(lambda[i]))

##PCAbiplot
ggplot2::autoplot(col0_phyD1.pca,label = F, loadings.label = T)
```


> From k-means clustering

```{r}
##EM clustering
library(mclust)
mclus_default<-Mclust(z)

mclass<-mclus_default$classification
k<-mclus_default$G
(mclus_default$modelName)

df <- data.frame(PC1 = z1, PC2 = z2, cluster = factor(mclass))

ggplot(df, aes(x = PC1, y = PC2, color = cluster)) +
  geom_point()
```


```{r}
length(phyD_c1[which(mclass==1),1])
#phyD_mclass1<-data.frame(phyD_c1[which(mclass==1),1])
#write.csv(phyD_mclass1, file = "phyD_mclass1.csv")

length(phyD_c1[which(mclass==2),1])
#phyD_mclass2<-data.frame(phyD_c1[which(mclass==2),1])
#write.csv(phyD_mclass2, file = "phyD_mclass2.csv")

length(phyD_c1[which(mclass==3),1])
#phyD_mclass3<-data.frame(phyD_c1[which(mclass==3),1])
#write.csv(phyD_mclass3, file = "phyD_mclass3.csv")

length(phyD_c1[which(mclass==4),1])
#phyD_mclass4<-data.frame(phyD_c1[which(mclass==4),1])
#write.csv(phyD_mclass4, file = "phyD_mclass4.csv")

length(phyD_c1[which(mclass==5),1])
#phyD_mclass5<-data.frame(phyD_c1[which(mclass==5),1])
#write.csv(phyD_mclass5, file = "phyD_mclass5.csv")

length(phyD_c1[which(mclass==6),1])
#phyD_mclass6<-data.frame(phyD_c1[which(mclass==6),1])
#write.csv(phyD_mclass6, file = "phyD_mclass6.csv")

length(phyD_c1[which(mclass==7),1])
#phyD_mclass7<-data.frame(phyD_c1[which(mclass==7),1])
#write.csv(phyD_mclass7, file = "phyD_mclass7.csv")

length(phyD_c1[which(mclass==8),1])
#phyD_mclass8<-data.frame(phyD_c1[which(mclass==8),1])
#write.csv(phyD_mclass8, file = "phyD_mclass8.csv")

length(phyD_c1[which(mclass==9),1])
#phyD_mclass9<-data.frame(phyD_c1[which(mclass==9),1])
#write.csv(phyD_mclass9, file = "phyD_mclass9.csv")
```

> From EM clustering

```{r}
#dbscan
library(dbscan)

kNNdistplot(z, k =  10)
abline(h = 50, lty = 2)

db<-dbscan(z,eps = 50,minPts = 10)
db


df <- data.frame(PC1 = z1, PC2 = z2, cluster = factor(db$cluster))


ggplot(df, aes(x = PC1, y = PC2, color = cluster)) +
  geom_point()

length(phyD_c1[which(db$cluster==1),1])
phyD_dbscan1<-data.frame(phyD_c1[which(db$cluster==1),1])

#write.csv(phyD_dbscan1, file = "phyD_dbscan1.csv")
```





```{r}
#phyD flight
#col0_phyD_F.pca<-princomp(col0_phyD[,c(2:7)])
#summary(col0_phyD_F.pca)
#z1<-col0_phyD_F.pca$scores[,1]
#z2<-col0_phyD_F.pca$scores[,2]
#z<-cbind(z1,z2)

#pcaplot <- data.frame(PC1 = z1, PC2 = z2)

#ggplot(pcaplot, aes(x = PC1, y = PC2)) +
#  geom_point()

#phyD GC
#col0_phyD_G.pca<-princomp(col0_phyD[,c(8:13)])
#summary(col0_phyD_G.pca)
#z1<-col0_phyD_G.pca$scores[,1]
#z2<-col0_phyD_G.pca$scores[,2]
#z<-cbind(z1,z2)

#pcaplot <- data.frame(PC1 = z1, PC2 = z2)

#ggplot(pcaplot, aes(x = PC1, y = PC2)) +
#  geom_point()


#phyD all light
#col0_phyD_light<-col0_phyD[,c(2:4,8:10)]
#col0_phyD_L.pca<-princomp(col0_phyD_light)
#summary(col0_phyD_L.pca)
#z1<-col0_phyD_L.pca$scores[,1]
#z2<-col0_phyD_L.pca$scores[,2]
#z<-cbind(z1,z2)

#pcaplot <- data.frame(PC1 = z1, PC2 = z2)

#ggplot(pcaplot, aes(x = PC1, y = PC2)) +
#  geom_point()

#phyD all dark
#col0_phyD_dark<-col0_phyD[,c(5:7,11:13)]
#col0_phyD_D.pca<-princomp(col0_phyD_dark)
#summary(col0_phyD_D.pca)
#z1<-col0_phyD_D.pca$scores[,1]
#z2<-col0_phyD_D.pca$scores[,2]
#z<-cbind(z1,z2)

#pcaplot <- data.frame(PC1 = z1, PC2 = z2)

#ggplot(pcaplot, aes(x = PC1, y = PC2)) +
#  geom_point()
```

```{r}
#col0
colnames(col0)<-c('A(1)','B(2)','C(3)','D(4)','E(5)','F(6)','G(7)','H(8)','I(9)','J(10)','K(11)','L(12)','M(13)')

#Has same outliers as col0_phyD
col0<-col0[-c(12958,23875,861),]
#PCA
col0.pca<-princomp(col0[,-1])
summary(col0.pca)
a1<-col0.pca$scores[,1]
a2<-col0.pca$scores[,2]
a<-cbind(a1,a2)

pcaplot <- data.frame(PC1 = a1, PC2 = a2)

ggplot(pcaplot, aes(x = PC1, y = PC2)) +
  geom_point()

eigvals<-col0.pca$sdev^2
k<-length(eigvals)
plot(1:k,eigvals,type="b",xlab="i",ylab=expression(lambda[i]))

##PCAbiplot
ggplot2::autoplot(col0.pca,label = F, loadings.label = T)

#which(a1>100000)
```

```{r}
##kmeans
n<-dim(a)[1]; k<-6
wss<- rep(0,k); xm<-apply(a,2,mean)
for(i in 1:n){
  wss[1]<- wss[1]+sum((a[i,]-xm)^2)
}
for(i in 2:k){
  model<-kmeans(a,i)
  wss[i]<-sum(model$withinss)
}
plot(1:k,wss,type="b",xlab="Number of clusters", ylab="Within cluster sum of squares",main="Screeplot")

k<-4
km4<-kmeans(a,k)

df <- data.frame(PC1 = a1, PC2 = a2, cluster = factor(km4$cluster))

ggplot(df, aes(x = PC1, y = PC2, color = cluster)) +
  geom_point()


length(col0[which(km4$cluster==1),1])

length(col0[which(km4$cluster==2),1])

length(col0[which(km4$cluster==3),1])

length(col0[which(km4$cluster==4),1])
```

```{r}
col_c1<-col0[which(km4$cluster==3),]
#PCA
col0_1.pca<-princomp(col_c1[,-1])
summary(col0_1.pca)
z1<-col0_1.pca$scores[,1]
z2<-col0_1.pca$scores[,2]
z<-cbind(z1,z2)

pcaplot <- data.frame(PC1 = z1, PC2 = z2)

ggplot(pcaplot, aes(x = PC1, y = PC2)) +
  geom_point()

eigvals<-col0_1.pca$sdev^2
k<-length(eigvals)
plot(1:k,eigvals,type="b",xlab="i",ylab=expression(lambda[i]))

##PCAbiplot
ggplot2::autoplot(col0_1.pca,label = F, loadings.label = T)
```

```{r}
##EM clustering
#library(mclust)
mclus_default<-Mclust(z)

mclass<-mclus_default$classification
k<-mclus_default$G
(mclus_default$modelName)

df <- data.frame(PC1 = z1, PC2 = z2, cluster = factor(mclass))

ggplot(df, aes(x = PC1, y = PC2, color = cluster)) +
  geom_point()
```

```{r}
length(col_c1[which(mclass==1),1])
col_mclass1<-data.frame(col_c1[which(mclass==1),1])
write.csv(col_mclass1, file = "col_mclass1.csv")

length(col_c1[which(mclass==2),1])
col_mclass2<-data.frame(col_c1[which(mclass==2),1])
write.csv(col_mclass2, file = "col_mclass2.csv")

length(col_c1[which(mclass==3),1])
col_mclass3<-data.frame(col_c1[which(mclass==3),1])
write.csv(col_mclass3, file = "col_mclass3.csv")

length(col_c1[which(mclass==4),1])
col_mclass4<-data.frame(col_c1[which(mclass==4),1])
write.csv(col_mclass4, file = "col_mclass4.csv")

length(col_c1[which(mclass==5),1])
col_mclass5<-data.frame(col_c1[which(mclass==5),1])
write.csv(col_mclass5, file = "col_mclass5.csv")

length(col_c1[which(mclass==6),1])
col_mclass6<-data.frame(col_c1[which(mclass==6),1])
write.csv(col_mclass6, file = "col_mclass6.csv")

length(col_c1[which(mclass==7),1])
col_mclass7<-data.frame(col_c1[which(mclass==7),1])
write.csv(col_mclass7, file = "col_mclass7.csv")

length(col_c1[which(mclass==8),1])
col_mclass8<-data.frame(col_c1[which(mclass==8),1])
write.csv(col_mclass8, file = "col_mclass8.csv")

length(col_c1[which(mclass==9),1])
col_mclass9<-data.frame(col_c1[which(mclass==9),1])
write.csv(col_mclass9, file = "col_mclass9.csv")
```


```{r}
#dbscan
library(dbscan)

kNNdistplot(z, k =  10)
abline(h = 50, lty = 2)

db<-dbscan(z,eps = 50,minPts = 10)
db


df <- data.frame(PC1 = z1, PC2 = z2, cluster = factor(db$cluster))


ggplot(df, aes(x = PC1, y = PC2, color = cluster)) +
  geom_point()

length(col_c1[which(db$cluster==1),1])
col_dbscan1<-data.frame(col_c1[which(db$cluster==1),1])

write.csv(col_dbscan1, file = "col_dbscan1.csv")
```


```{r}
#WS
colnames(WS)<-c('A(1)','B(2)','C(3)','D(4)','E(5)','F(6)','G(7)','H(8)','I(9)','J(10)','K(11)','L(12)','M(13)')

#Has same outliers as col0_phyD
WS<-WS[-c(12958,23875,861),]
#PCA
WS.pca<-princomp(WS[,-1])
summary(WS.pca)
a1<-WS.pca$scores[,1]
a2<-WS.pca$scores[,2]
a<-cbind(a1,a2)

pcaplot <- data.frame(PC1 = a1, PC2 = a2)

ggplot(pcaplot, aes(x = PC1, y = PC2)) +
  geom_point()

eigvals<-WS.pca$sdev^2
k<-length(eigvals)
plot(1:k,eigvals,type="b",xlab="i",ylab=expression(lambda[i]))

##PCAbiplot
ggplot2::autoplot(WS.pca,label = F, loadings.label = T)

#which(a1>100000)
```

```{r}
##kmeans
n<-dim(a)[1]; k<-6
wss<- rep(0,k); xm<-apply(a,2,mean)
for(i in 1:n){
  wss[1]<- wss[1]+sum((a[i,]-xm)^2)
}
for(i in 2:k){
  model<-kmeans(a,i)
  wss[i]<-sum(model$withinss)
}
plot(1:k,wss,type="b",xlab="Number of clusters", ylab="Within cluster sum of squares",main="Screeplot")

k<-4
km4<-kmeans(a,k)

df <- data.frame(PC1 = a1, PC2 = a2, cluster = factor(km4$cluster))

ggplot(df, aes(x = PC1, y = PC2, color = cluster)) +
  geom_point()


length(WS[which(km4$cluster==1),1])

length(WS[which(km4$cluster==2),1])

length(WS[which(km4$cluster==3),1])

length(WS[which(km4$cluster==4),1])
```


```{r}
WS_c1<-WS[which(km4$cluster==1),]
#PCA
WS_1.pca<-princomp(WS_c1[,-1])
summary(WS_1.pca)
z1<-WS_1.pca$scores[,1]
z2<-WS_1.pca$scores[,2]
z<-cbind(z1,z2)

pcaplot <- data.frame(PC1 = z1, PC2 = z2)

ggplot(pcaplot, aes(x = PC1, y = PC2)) +
  geom_point()

eigvals<-WS_1.pca$sdev^2
k<-length(eigvals)
plot(1:k,eigvals,type="b",xlab="i",ylab=expression(lambda[i]))

##PCAbiplot
ggplot2::autoplot(WS_1.pca,label = F, loadings.label = T)
```

```{r}
##EM clustering
#library(mclust)
mclus_default<-Mclust(z)

mclass<-mclus_default$classification
k<-mclus_default$G
(mclus_default$modelName)

df <- data.frame(PC1 = z1, PC2 = z2, cluster = factor(mclass))

ggplot(df, aes(x = PC1, y = PC2, color = cluster)) +
  geom_point()
```

```{r}
length(WS_c1[which(mclass==1),1])
WS_mclass1<-data.frame(WS_c1[which(mclass==1),1])
write.csv(WS_mclass1, file = "WS_mclass1.csv")

length(WS_c1[which(mclass==2),1])
WS_mclass2<-data.frame(WS_c1[which(mclass==2),1])
write.csv(WS_mclass2, file = "WS_mclass2.csv")

length(WS_c1[which(mclass==3),1])
WS_mclass3<-data.frame(WS_c1[which(mclass==3),1])
write.csv(WS_mclass3, file = "WS_mclass3.csv")

length(WS_c1[which(mclass==4),1])
WS_mclass4<-data.frame(WS_c1[which(mclass==4),1])
write.csv(WS_mclass4, file = "WS_mclass4.csv")

length(WS_c1[which(mclass==5),1])
WS_mclass5<-data.frame(WS_c1[which(mclass==5),1])
write.csv(WS_mclass5, file = "WS_mclass5.csv")

length(WS_c1[which(mclass==6),1])
WS_mclass6<-data.frame(WS_c1[which(mclass==6),1])
write.csv(WS_mclass6, file = "WS_mclass6.csv")

length(WS_c1[which(mclass==7),1])
WS_mclass7<-data.frame(WS_c1[which(mclass==7),1])
write.csv(WS_mclass7, file = "WS_mclass7.csv")

length(WS_c1[which(mclass==8),1])
WS_mclass8<-data.frame(WS_c1[which(mclass==8),1])
write.csv(WS_mclass8, file = "WS_mclass8.csv")
```

```{r}
#dbscan
#library(dbscan)

kNNdistplot(z, k =  10)
abline(h = 50, lty = 2)

db<-dbscan(z,eps = 50,minPts = 10)
db


df <- data.frame(PC1 = z1, PC2 = z2, cluster = factor(db$cluster))


ggplot(df, aes(x = PC1, y = PC2, color = cluster)) +
  geom_point()

length(WS_c1[which(db$cluster==1),1])
WS_dbscan1<-data.frame(WS_c1[which(db$cluster==1),1])

write.csv(WS_dbscan1, file = "WS_dbscan1.csv")
```

```{r}
WS_c2<-WS[which(db$cluster==1),]
#PCA
WS_2.pca<-princomp(WS_c2[,-1])
summary(WS_2.pca)
z1<-WS_2.pca$scores[,1]
z2<-WS_2.pca$scores[,2]
z<-cbind(z1,z2)

pcaplot <- data.frame(PC1 = z1, PC2 = z2)

ggplot(pcaplot, aes(x = PC1, y = PC2)) +
  geom_point()

eigvals<-WS_2.pca$sdev^2
k<-length(eigvals)
plot(1:k,eigvals,type="b",xlab="i",ylab=expression(lambda[i]))

##PCAbiplot
ggplot2::autoplot(WS_2.pca,label = F, loadings.label = T)
```

??? Feels like I'm in a loop hole
No specific pattern or distinct cluster

---> needs new approach! 
Change the structure of the dataset

