
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  </head>
<body ng-app="app">
<div class="container">
        <h2>Metascape Gene List Analysis Report</h2><p/>  <a href="http://metascape.org">metascape.org</a><sup>1</sup><p/>

              <h3>Bar Graph Summary</h3>
              <div class="panel panel-info">
                <div class="panel-heading">Figure 1. Bar graph of enriched terms across input gene lists, colored by p-values.</div>
                <div class="panel-body">

                <table>
                    <tr>
                        <td>
                            <img src="./Enrichment_heatmap/HeatmapSelectedGO.png" style="width:1000px;">
                        </td>
                    </tr>
                    <tr>
                        <td align='center'>
                            
                <a href='./Enrichment_heatmap/HeatmapSelectedGO.pdf' title='download PDF file'>
                <img class='link' src='icon/PDF48.png' >
                </a>
                
                        </td>
                    </tr>
                    
                <tr><td>Metascape only visualizes the top 20 clusters. Up to 100 enriched clusters can be viewed here.<br>
                <a href="./Enrichment_heatmap/HeatmapSelectedGOTop100.png"><img src="./Enrichment_heatmap/HeatmapSelectedGOTop100.png" style="height:auto; width:auto; max-height:100px;"></a>
                <a href='./Enrichment_heatmap/HeatmapSelectedGOTop100.pdf' title='download PDF file'>
                <img class='link' src='icon/PDF48.png' >
                </a></td></tr>
                
                <tr><td>The top-level Gene Ontology biological processes can be viewed here.<br>
                <a href="./Enrichment_heatmap/HeatmapSelectedGOParent.png"><img src="./Enrichment_heatmap/HeatmapSelectedGOParent.png" style="height:auto; width:auto; max-height:100px;"></a>
                <a href='./Enrichment_heatmap/HeatmapSelectedGOParent.pdf' title='download PDF file'>
                <img class='link' src='icon/PDF48.png' >
                </a></td></tr>
                
                </table>
                </div>
              </div><p/>
      <h3>Gene Lists</h3>
      User-provided gene identifiers are first converted into their corresponding A. thaliana Entrez gene IDs using the latest version of the database (last updated on 2020-09-16).  If multiple identifiers correspond to the same Entrez gene ID, they will be considered as a single Entrez gene ID in downstream analyses.   The gene lists are summarized in Table 1.<p/>
          <div class="panel panel-info">
            <div class="panel-heading">Table 1. Statistics of input gene lists.</div>
            <div class="panel-body"><TABLE class="table">
  <THEAD>
    <TR>
      <TH class="info">Name</TH>
      <TH class="info">Total</TH>
      <TH class="info">Unique</TH>
    </TR>
  </THEAD>
  <TBODY>
    <TR>
      <TD>hits</TD>
      <TD>3000</TD>
      <TD>2995</TD>
    </TR>
  </TBODY>
</TABLE></div>
          </div>


              <h3>Gene Annotation</h3>
              The following are the list of annotations retrieved from the latest version of the database (last updated on 2020-09-16) (Table 2).<p/>
              <div class="panel panel-info">
                <div class="panel-heading">Table 2. Gene annotations extracted</div>
                <div class="panel-body"><TABLE class="table">
  <THEAD>
    <TR>
      <TH class="info">Name</TH>
      <TH class="info">Type</TH>
      <TH class="info">Description</TH>
    </TR>
  </THEAD>
  <TBODY>
    <TR>
      <TD>Gene Symbol</TD>
      <TD>Description</TD>
      <TD>Primary HUGO gene symbol.</TD>
    </TR>
    <TR>
      <TD>Description</TD>
      <TD>Description</TD>
      <TD>Short description.</TD>
    </TR>
    <TR>
      <TD>Biological Process (GO)</TD>
      <TD>Function/Location</TD>
      <TD>Descriptions summarized based on gene ontology database, where up to three most informative GO terms are kept.</TD>
    </TR>
    <TR>
      <TD>Kinase Class (UniProt)</TD>
      <TD>Function/Location</TD>
      <TD>Detailed kinase classes.</TD>
    </TR>
    <TR>
      <TD>Protein Function (Protein Atlas)</TD>
      <TD>Function/Location</TD>
      <TD>Protein Function (Protein Atlas)</TD>
    </TR>
    <TR>
      <TD>Subcellular Location (Protein Atlas)</TD>
      <TD>Function/Location</TD>
      <TD>Sucellular Location (Protein Atlas)</TD>
    </TR>
    <TR>
      <TD>Drug (DrugBank)</TD>
      <TD>Genotype/Phenotype/Disease</TD>
      <TD>Drug information for the given gene as target.</TD>
    </TR>
    <TR>
      <TD>Canonical Pathways
</TD>
      <TD>Ontology</TD>
      <TD>Canonical Pathways
</TD>
    </TR>
    <TR>
      <TD>Hallmark Gene Sets
</TD>
      <TD>Ontology</TD>
      <TD>Hallmark Gene Sets
</TD>
    </TR>
  </TBODY>
</TABLE></div>
              </div><p/>


              <h3>Pathway and Process Enrichment Analysis</h3>
              For each given gene list, pathway and process enrichment analysis has been carried out with the following ontology sources: GO Biological Processes and KEGG Pathway.  All genes in the genome have been used as the enrichment background. Terms with a p-value &lt; 0.01, a minimum count of 3, and an enrichment factor &gt; 1.5 (the enrichment factor is the ratio between the observed counts and the counts expected by chance) are collected and grouped into clusters based on their membership similarities.  More specifically, p-values are calculated based on the accumulative hypergeometric distribution<sup>2</sup>, and q-values are calculated using the Banjamini-Hochberg procedure to account for multiple testings<sup>3</sup>.  Kappa scores<sup>4</sup> are used as the similarity metric when performing hierachical clustering on the enriched terms, and sub-trees with a similarity of > 0.3 are considered a cluster.  The most statistically significant term within a cluster is chosen to represent the cluster.<p/>
          <div class="panel panel-info">
          <div class="panel-heading">Table 3. Top 20 clusters with their representative enriched terms (one per cluster).  "Count" is the number of genes in the user-provided lists with membership in the given ontology term. "%" is the percentage of all of the user-provided genes that are found in the given ontology term (only input genes with at least one ontology term annotation are included in the calculation).  "Log10(P)" is the p-value in log base 10.  "Log10(q)" is the multi-test adjusted p-value in log base 10.</div>
                <div class="panel-body"><TABLE class="table">
  <THEAD>
    <TR>
      <TH class="info">GO</TH>
      <TH class="info">Category</TH>
      <TH class="info">Description</TH>
      <TH class="info">Count</TH>
      <TH class="info">%</TH>
      <TH class="info">Log10(P)</TH>
      <TH class="info">Log10(q)</TH>
    </TR>
  </THEAD>
  <TBODY>
    <TR>
      <TD>GO:0046907</TD>
      <TD>GO Biological Processes</TD>
      <TD>intracellular transport</TD>
      <TD>154</TD>
      <TD>6.52</TD>
      <TD>-13.49</TD>
      <TD>-9.68</TD>
    </TR>
    <TR>
      <TD>GO:0006520</TD>
      <TD>GO Biological Processes</TD>
      <TD>cellular amino acid metabolic process</TD>
      <TD>104</TD>
      <TD>4.40</TD>
      <TD>-10.94</TD>
      <TD>-7.61</TD>
    </TR>
    <TR>
      <TD>GO:0016192</TD>
      <TD>GO Biological Processes</TD>
      <TD>vesicle-mediated transport</TD>
      <TD>111</TD>
      <TD>4.70</TD>
      <TD>-10.70</TD>
      <TD>-7.58</TD>
    </TR>
    <TR>
      <TD>GO:0048583</TD>
      <TD>GO Biological Processes</TD>
      <TD>regulation of response to stimulus</TD>
      <TD>144</TD>
      <TD>6.10</TD>
      <TD>-6.89</TD>
      <TD>-4.33</TD>
    </TR>
    <TR>
      <TD>ath03008</TD>
      <TD>KEGG Pathway</TD>
      <TD>Ribosome biogenesis in eukaryotes</TD>
      <TD>32</TD>
      <TD>1.35</TD>
      <TD>-6.85</TD>
      <TD>-4.33</TD>
    </TR>
    <TR>
      <TD>GO:0010015</TD>
      <TD>GO Biological Processes</TD>
      <TD>root morphogenesis</TD>
      <TD>66</TD>
      <TD>2.79</TD>
      <TD>-6.65</TD>
      <TD>-4.17</TD>
    </TR>
    <TR>
      <TD>ath03040</TD>
      <TD>KEGG Pathway</TD>
      <TD>Spliceosome</TD>
      <TD>49</TD>
      <TD>2.07</TD>
      <TD>-6.34</TD>
      <TD>-3.93</TD>
    </TR>
    <TR>
      <TD>GO:0070646</TD>
      <TD>GO Biological Processes</TD>
      <TD>protein modification by small protein removal</TD>
      <TD>28</TD>
      <TD>1.19</TD>
      <TD>-5.71</TD>
      <TD>-3.32</TD>
    </TR>
    <TR>
      <TD>GO:1901361</TD>
      <TD>GO Biological Processes</TD>
      <TD>organic cyclic compound catabolic process</TD>
      <TD>63</TD>
      <TD>2.67</TD>
      <TD>-5.54</TD>
      <TD>-3.19</TD>
    </TR>
    <TR>
      <TD>GO:0015931</TD>
      <TD>GO Biological Processes</TD>
      <TD>nucleobase-containing compound transport</TD>
      <TD>33</TD>
      <TD>1.40</TD>
      <TD>-5.25</TD>
      <TD>-2.98</TD>
    </TR>
    <TR>
      <TD>GO:0043648</TD>
      <TD>GO Biological Processes</TD>
      <TD>dicarboxylic acid metabolic process</TD>
      <TD>27</TD>
      <TD>1.14</TD>
      <TD>-5.20</TD>
      <TD>-2.96</TD>
    </TR>
    <TR>
      <TD>GO:0030490</TD>
      <TD>GO Biological Processes</TD>
      <TD>maturation of SSU-rRNA</TD>
      <TD>18</TD>
      <TD>0.76</TD>
      <TD>-5.07</TD>
      <TD>-2.85</TD>
    </TR>
    <TR>
      <TD>GO:0071806</TD>
      <TD>GO Biological Processes</TD>
      <TD>protein transmembrane transport</TD>
      <TD>29</TD>
      <TD>1.23</TD>
      <TD>-5.01</TD>
      <TD>-2.81</TD>
    </TR>
    <TR>
      <TD>GO:0010315</TD>
      <TD>GO Biological Processes</TD>
      <TD>auxin efflux</TD>
      <TD>9</TD>
      <TD>0.38</TD>
      <TD>-4.76</TD>
      <TD>-2.61</TD>
    </TR>
    <TR>
      <TD>ath00230</TD>
      <TD>KEGG Pathway</TD>
      <TD>Purine metabolism</TD>
      <TD>38</TD>
      <TD>1.61</TD>
      <TD>-4.45</TD>
      <TD>-2.38</TD>
    </TR>
    <TR>
      <TD>GO:0006515</TD>
      <TD>GO Biological Processes</TD>
      <TD>protein quality control for misfolded or incompletely synthesized proteins</TD>
      <TD>13</TD>
      <TD>0.55</TD>
      <TD>-4.41</TD>
      <TD>-2.35</TD>
    </TR>
    <TR>
      <TD>GO:0010646</TD>
      <TD>GO Biological Processes</TD>
      <TD>regulation of cell communication</TD>
      <TD>69</TD>
      <TD>2.92</TD>
      <TD>-4.40</TD>
      <TD>-2.34</TD>
    </TR>
    <TR>
      <TD>GO:0044282</TD>
      <TD>GO Biological Processes</TD>
      <TD>small molecule catabolic process</TD>
      <TD>51</TD>
      <TD>2.16</TD>
      <TD>-4.29</TD>
      <TD>-2.25</TD>
    </TR>
    <TR>
      <TD>GO:0018193</TD>
      <TD>GO Biological Processes</TD>
      <TD>peptidyl-amino acid modification</TD>
      <TD>70</TD>
      <TD>2.96</TD>
      <TD>-4.13</TD>
      <TD>-2.14</TD>
    </TR>
    <TR>
      <TD>GO:0017144</TD>
      <TD>GO Biological Processes</TD>
      <TD>drug metabolic process</TD>
      <TD>114</TD>
      <TD>4.83</TD>
      <TD>-4.09</TD>
      <TD>-2.13</TD>
    </TR>
  </TBODY>
</TABLE></div>
              </div><p/>
                  To further capture the relationships between the terms, a subset of enriched terms have been selected and rendered as a network plot, where terms with a similarity &gt; 0.3 are connected by edges.  We select the terms with the best p-values from each of the 20 clusters, with the constraint that there are no more than 15 terms per cluster and no more than 250 terms in total.  The network is visualized using <a href="http://www.cytoscape.org">Cytoscape</a><sup>5</sup>, where each node represents an enriched term and is colored first by its cluster ID (Figure 2.a) and then by its p-value (Figure 2.b).  These networks can be interactively viewed in Cytoscape through the .cys files (contained in the Zip package, which also contains a publication-quality version as a PDF) or within a browser by clicking on the web icon.  For clarity, term labels are only shown for one term per cluster, so it is recommended to use Cytoscape or a browser to visualize the network in order to inspect all node labels.  We can also export the network into a PDF file within Cytoscape, and then edit the labels using Adobe Illustrator for publication purposes.  To switch off all labels, delete the "Label" mapping under the "Style" tab within Cytoscape, and then export the network view.<p/>
                  <div class="panel panel-info">
                    <div class="panel-heading">Figure 2. Network of enriched terms: (a) colored by cluster ID, where nodes that share the same cluster ID are typically close to each other; (b) colored by p-value, where terms containing more genes tend to have a more significant p-value.</div>
                    <div class="panel-body">
                    <table>
                        <tr>
                            <td>
                                <img src="./Enrichment_GO/ColorByCluster.png" style="width:500px;">
                            </td>
                            <td>
                                <img src="./Enrichment_GO/ColorByPValue.png" style="width:500px;">
                            </td>
                        </tr>
                        <tr>
                            <td align='center'>
                                <a href='./Enrichment_GO/ColorByCluster.pdf' title='download PDF file'>
                        <img class='link' src='icon/PDF48.png' >
                        </a>
                     
                        <a href='./Enrichment_GO/GONetwork.cys' title='download CYS file'>
                        <img class='link' src='icon/CYS48.png' >
                        </a>
                    
                        
                    
                        
                        <a href='Enrichment_GO/GONetwork.html?Network=GONetwork&Style=ColorByCluster' title='interactive cytoscape' target='_blank'>
                        <img class='link' src='icon/WEB_CYS48.png' >
                        </a>
                        
                    
                            </td>
                            <td align='center'>
                                <a href='./Enrichment_GO/ColorByPValue.pdf' title='download PDF file'>
                        <img class='link' src='icon/PDF48.png' >
                        </a>
                     
                        <a href='./Enrichment_GO/GONetwork.cys' title='download CYS file'>
                        <img class='link' src='icon/CYS48.png' >
                        </a>
                    
                        
                    
                        
                        <a href='Enrichment_GO/GONetwork.html?Network=GONetwork&Style=ColorByPValue' title='interactive cytoscape' target='_blank'>
                        <img class='link' src='icon/WEB_CYS48.png' >
                        </a>
                        
                    
                            </td>
                        </tr>
                    </table>
                    </div>

                  </div><p/>
<h3>Protein-protein Interaction Enrichment Analysis</h3>
For each given gene list, protein-protein interaction enrichment analysis has been carried out with the following databases: BioGrid<sup>6</sup>. The resultant network contains the subset of proteins that form physical interactions with at least one other member in the list.  If the network contains between 3 and 500 proteins, the Molecular Complex Detection (MCODE) algorithm<sup>7</sup> has been applied to identify densely connected network components.  


    <h3>Reference</h3></p>
    <ol style="list-style: decimal inside;">
      <li>Zhou et al., Metascape provides a biologist-oriented resource for the analysis of systems-level datasets. Nature Communications (2019) 10(1):1523.</li>
      <li>Zar, J.H. Biostatistical Analysis 1999 4th edn., NJ Prentice Hall, pp. 523</li>
      <li>Hochberg Y., Benjamini Y. More powerful procedures for multiple significance testing. Statistics in Medicine (1990) 9:811-818.</li>
      <li>Cohen, J. A coefficient of agreement for nominal scales. Educ. Psychol. Meas. (1960) 20:27-46.</li>
      <li>Shannon P. et al., Cytoscape: a software environment for integrated models of biomolecular interaction networks. Genome Res (2003) 11:2498-2504.</li>
      <li>Stark C. et al. BioGRID: a general repository for interaction datasets. Nucleic Acids Res. (2006) 34:D535-539.</li>
      <li>Bader, G.D. et al. An automated method for finding molecular complexes in large protein interaction networks. BMC bioinformatics (2003) 4:2.</li>
</ol>
</div>

</body></html>